File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int res=-1;
        int zero=0;
        int l=0;
        for(int r=0;r<nums.length;r++){
            if (nums[r]==0){
                zero++;
            }
            while(zero > k){
                if (nums[l]==0){
                    zero--;
                }
                l++;
            }
            res=Math.max(res,r-l+1);
        }

        return res;
        
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        res=-1
        zero=0

        l=0
        for r in range(len(nums)):
            if nums[r]==0:
                zero+=1
            while zero > k:
                if nums[l]==0:
                    zero-=1
                l+=1
            res=max(res,r-l+1)
        return res 
        